android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -244,12 +248,31 @@ public void handleMessage(Message msg) {
                         }
                         msg = obtainMessage(SHOW_PROGRESS);
                         sendMessageDelayed(msg, Math.round(mProgressUpdateInterval));
+                        setFrame();
                     }
                     break;
             }
         }
     };
 
+    private void setFrame() {
+        if(frameQuality == null) return;
+        if(frameQuality > 1) frameQuality = 1;
+        frameQuality = Math.round(frameQuality * 100);
+        ((Runnable) () -> {
```

Comment:
Are you sure Runnable.run() will really delay the call ?
I think you should use a handler to do that. else it will just run here

Reply:
Runnable won't delay but It doesn't block main thread or ui thread